tests/e2e/support/objects/app-files/share/collaborator.ts

```diff
@@ -0,0 +1,227 @@
+  ): Promise<void> {
+    for (const permission of permissions) {
+      if (!Collaborator.CUSTOM_PERMISSIONS.includes(permission)) {
+        throw new Error(
+          'Invalid custom permission: ' +
+            permission +
+            '\nAvailable permissions: ' +
+            Collaborator.CUSTOM_PERMISSIONS
+        )
+      }
+      await page.check(util.format(Collaborator.customPermissionCheckbox, permission))
+    }
+    // uncheck others
+    const removePermissions = difference(Collaborator.CUSTOM_PERMISSIONS, permissions)
+    for (const permission of removePermissions) {
+      await page.uncheck(util.format(Collaborator.customPermissionCheckbox, permission))
+    }
+  }
+
+  static async setCollaboratorRole(
+    page: Page,
+    role: string,
+    dropdownSelector?: string,
+    itemSelector?: string
+  ): Promise<void> {
+    if (!dropdownSelector) {
+      dropdownSelector = Collaborator.newCollaboratorRoleDropdown
+      itemSelector = Collaborator.newCollaboratorRoleItemSelector
+    }
+    await page.click(dropdownSelector)
+
+    // custom permissions should be set as below
+    // custom_permissions:read,share
+    if (role.includes('custom_permissions')) {
+      await page.click(util.format(itemSelector, 'custom'))
+      const custom_permissions = role.split(':')[1]
+
+      if (!custom_permissions) {
+        throw new Error('No custom permissions provided: ' + custom_permissions)
+      }
+
+      const permissions = custom_permissions.split(',')
+      await Collaborator.setCustomPermissions(page, permissions as CustomPermissionType[])
+
+      return await page.click(Collaborator.customPermissionApplyButton)
+    }
+    return await page.click(util.format(itemSelector, role))
+  }
+
+  static async changeCollaboratorRole(args: CollaboratorArgs): Promise<void> {
+    const {
+      page,
+      collaborator: { collaborator, type, role }
+    } = args
+
+    const collaboratorRow = Collaborator.getCollaboratorUserOrGroupSelector(collaborator, type)
+    const roleDropdownSelector = util.format(
+      Collaborator.collaboratorRoleDropdownButton,
+      collaboratorRow
+    )
+    const roleItemSelector = util.format(Collaborator.collaboratorRoleItemSelector, collaboratorRow)
+    await Collaborator.setCollaboratorRole(page, role, roleDropdownSelector, roleItemSelector)
+  }
+
+  static async removeCollaborator(args: RemoveCollaboratorArgs): Promise<void> {
+    const {
+      page,
+      collaborator: { collaborator, type },
+      removeOwnSpaceAccess
+    } = args
+    const collaboratorRow = Collaborator.getCollaboratorUserOrGroupSelector(collaborator, type)
+
+    await Promise.all([
+      page.waitForResponse(
+        (resp) =>
+          resp.url().includes('shares') &&
+          resp.status() === 200 &&
+          resp.request().method() === 'DELETE'
+      ),
+      page
+        .locator(util.format(Collaborator.collaboratorEditDropdownButton, collaboratorRow))
+        .first() // remove
```

Comment:
need to be removed?

Reply:
Oh yeah, I need to check what that does.